### ISSUE TITLE
Pep 728 support for TypedDict with Unpack does not allow extra items

### ISSUE THREAD

**Issue by gazpachoking** [microsoft/pyright#10352]
**Describe the bug**
When typing kwargs with an `Unpack`ed `TypedDict` that has `extra_items` defined, extra kwargs are not actually allowed when calling the function.

**Code or Screenshots**

The issue can be seen perfectly here, and there are comments saying that it's ambiguous whether this should be allowed. (I think line 34 should not cause any errors)

https://github.com/microsoft/pyright/blob/1d4663540fc675887a56a1501fbb1bc53c6fc37d/packages/pyright-internal/src/tests/samples/typedDictClosed2.py#L18-L34

My reading of PEP 728 has support for this as one of the [stated goals.](https://peps.python.org/pep-0728/#support-additional-keys-for-unpack)

In the [interaction with Unpack](https://peps.python.org/pep-0728/#interaction-with-unpack) section it makes it more clear. Here's how I think that section should apply to the example.

```python
class Movie2(TypedDict, extra_items=int): 
    name: str 
  
  
def func2(**kwargs: Unpack[Movie2]) -> None: ... 

# this should be equivalent to
def func2(*, name: str, **kwargs: int) -> None: ...
```
**VS Code extension or command-line**
Are you running pyright as a VS Code extension, a language server in another editor, integrated into Pylance, or the command-line tool? Which version?

Running 1.1.399 from command line.

**Comment by erictraut** [comment]
The intended behavior isn't as clear to me from reading the current draft of the PEP, but I think you have made a compelling argument.

It would be good for the PEP to be more explicit about this case because it does appear to contradict the behavior described in the earlier PEP 692.

**Comment by erictraut** [comment]
This is addressed in 1.1.400.
